mlkem/poly.h

```diff
@@ -209,7 +209,7 @@ void poly_frombytes(poly *r, const uint8_t a[MLKEM_POLYBYTES])
     // clang-format off
 REQUIRES(IS_FRESH(a, MLKEM_POLYBYTES))
 REQUIRES(IS_FRESH(r, sizeof(poly)))
-ASSIGNS(OBJECT_WHOLE(r))
+ASSIGNS(OBJECT_UPTO(r, sizeof(poly)))
```

Comment:
Can you explain this change to me?

Reply:
With `OBJECT_WHOLE`, the entire object is marked as overwritten -- if `r` is the pointer to a `poly` inside a `poly[]`, that would taint the whole array. Because of this, the `polyvec*` functions can't reason anymore that each iteration only changes one entry, not all of them. With `OBJECT_UPTO`, we indicate that only a slice of the specific size is invalidated, which gives CBMC the means to reason that each iteration modifies only one poly.